quickwit/quickwit-proto/src/types/mod.rs

```diff
@@ -47,83 +49,10 @@ pub fn queue_id(index_uid: &str, source_id: &str, shard_id: u64) -> QueueId {
 
 pub fn split_queue_id(queue_id: &str) -> Option<(IndexUid, SourceId, ShardId)> {
     let mut parts = queue_id.split('/');
-    let index_uid = parts.next()?;
+    let index_uid: IndexUid = parts.next()?.parse().ok()?;
     let source_id = parts.next()?;
     let shard_id = parts.next()?.parse::<u64>().ok()?;
-    Some((index_uid.into(), source_id.to_string(), shard_id))
-}
-
-/// Index identifiers that uniquely identify not only the index, but also
-/// its incarnation allowing to distinguish between deleted and recreated indexes.
-/// It is represented as a stiring in index_id:incarnation_id format.
-#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq, Eq, Ord, PartialOrd, Hash)]
-pub struct IndexUid(String);
-
-impl fmt::Display for IndexUid {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{}", self.0)
-    }
-}
-
-impl IndexUid {
-    /// Creates a new index uid form index_id and incarnation_id
-    pub fn new(index_id: impl Into<String>) -> Self {
-        Self::from_parts(index_id, Ulid::new().to_string())
-    }
-
-    pub fn as_str(&self) -> &str {
-        &self.0
-    }
-
-    pub fn from_parts(index_id: impl Into<String>, incarnation_id: impl Into<String>) -> Self {
-        let incarnation_id = incarnation_id.into();
-        let index_id = index_id.into();
-        if incarnation_id.is_empty() {
```

Comment:
This is needed for backward compatibility. We should no longer need this as we're dropping backward compatibility with 0.4 and 0.5 in the next release.